os/mm/mm_heap/mm_realloc.c

```diff
@@ -136,6 +136,13 @@ FAR void *mm_realloc(FAR struct mm_heap_s *heap, FAR void *oldmem, size_t size)
 		return NULL;
 	}
 
+	if (size > MM_ALIGN_DOWN(MMSIZE_MAX) - SIZEOF_MM_ALLOCNODE) {
+		mdbg("Because of mm_allocnode, %u cannot be allocated. The maximum \
+			 allocable size is (MM_ALIGN_DOWN(MMSIZE_MAX) - SIZEOF_MM_ALLOCNODE) \
+			 : %u\n.", size, (MM_ALIGN_DOWN(MMSIZE_MAX) - SIZEOF_MM_ALLOCNODE));
+		return NULL;
```

Comment:
Why don't you free 'oldmem' before return?

Reply:
I think `free` is not needed, because 
1) if `oldmem` allocates well, then already returned from line 126 or 128.
2) even if `oldmem` is not null, we don't need to free the original memory. free is only working when requested size is 0.(line 134)

Reply:
Abnormal requests need to be treated with some penalties.
Anyhow, you should describe how to treat this case.

Reply:
Please, check POSIX. We need to follow the standard.

Reply:
There is a guide in POSIX documents like below.
```
If the space cannot be allocated, the object shall remain unchanged.
```
So I think we don't need to free the original.
And I think, regardless of POSIX description, `realloc` is for re-allocating new memory with new size. If we cannot allocate for any reason, we should not spoil the original memory.